components/server/src/express/ws-handler.ts

```diff
@@ -53,11 +56,26 @@ export class WsExpressHandler {
         this.wss.on("error", (err) => {
             log.error("websocket server error", err, { wss: this.wss });
         });
+        this.wss.on("connection", (ws) => {
+            const cancelTerminate = this.disposables.push(Disposable.create(() => ws.terminate()));
+            ws.on("close", () => {
+                cancelTerminate.dispose();
+            });
+        });
         this.httpServer.on("upgrade", (req: http.IncomingMessage, socket: net.Socket, head: Buffer) =>
             this.onUpgrade(req, socket, head),
         );
     }
 
+    dispose(): void {
+        this.wss.close((err) => {
```

Comment:
We do similar bookkeeping here: https://github.com/gitpod-io/gitpod/blob/478a75e744a642d9b764de37cfae655bc8b29dd5/components/server/src/express/ws-connection-handler.ts#L21.
We could register the disposables there as well.